### PR TITLE
fix: add table id to Datanode heartbeat

### PR DESCRIPTION
**Pull request by tarunsamanta2k20**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?
What type of enhancement is this?
Refactor

What does the enhancement do?
Table id is just as Important as its name, we should carry it Datanode's heartbeat. It's relatively simple to do, table id is stored in table meta, which is already present when assembling heartbeat data.

Implementation challenges
No response
_PLEASE DO NOT LEAVE THIS EMPTY !!!_

Please explain IN DETAIL what the changes are in this PR and why they are needed:

- Summarize your change (**mandatory**)
- How does this PR work? Need a brief introduction for the changed logic (optional)
- Describe clearly one logical change and avoid lazy messages (optional)
- Describe any limitations of the current code (optional)

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

### CODE REVIEW

**Comment by tarunsamanta2k20** [comment]
@MichaelScofield I believe that some modifications are required. Thank you. Please check once and let me know.

**Comment by tarunsamanta2k20** [comment]
@MichaelScofield I believe that some modifications are required. Thank you. Please check once and let me know. It was from my guess that I did.

**Comment by tarunsamanta2k20** [comment]
@MichaelScofield I believe that some modifications are required. Thank you. Please check once and let me know. It was from my guess that I did.

**Comment by tarunsamanta2k20** [comment]
@MichaelScofield first finalizes the changes, later I will modify PR description.

**Comment by tarunsamanta2k20** [comment]
@MichaelScofield let me correct basically there were some changes I forgot to take pull. My branch is behind.

File: src/catalog/src/lib.rs [hunk 1]
```diff
@@ -106,6 +107,7 @@ pub trait CatalogManager: Send + Sync {
         catalog: &str,
         schema: &str,
         table_name: &str,
+        table_id: &uint64,
```
Comment (MichaelScofield) [inline src/catalog/src/lib.rs:110]: I think `table_id` should not put here.

**Review by MichaelScofield — COMMENTED** [review]
@tarunsamanta2k20 there are still some compiler errors

File: src/catalog/src/lib.rs (hunk 1 above)
Comment (tarunsamanta2k20) [reply src/catalog/src/lib.rs:110]: @MichaelScofield should 
I remove it from there?

Comment (MichaelScofield) [reply src/catalog/src/lib.rs:110]: yes

**Comment by tarunsamanta2k20** [comment]
@MichaelScofield there might be other changes as well.

**Comment by MichaelScofield** [comment]
@tarunsamanta2k20 you can run `cargo build` to see the compiler errors

**Comment by tarunsamanta2k20** [comment]
> @tarunsamanta2k20 you can run `cargo build` to see the compiler errors.

What's wrong in the below code?

 **let Ok(table_ids) = schema.table_ids().await else { continue };
                for table_id in table_ids {
                    let Ok(Some(table)) = schema.table(&table_id).await else { continue };**

**Comment by tarunsamanta2k20** [comment]
@MichaelScofield don't run check now,. I have to verify.

**Comment by tarunsamanta2k20** [comment]
@MichaelScofield is there any way to get the table_id directly from table_name I think no need of looping as one table_name can have only one table_id. Correct me if I am wrong.

**Comment by MichaelScofield** [comment]
@tarunsamanta2k20 this feature is implemented elsewhere, thank for your effort!

**Pull request closed without merging**